### PR TITLE
WS-71: Edited SightingsCard

### PR DESCRIPTION
**Pull request by tomscambler**
Added in a date section for the sightings card - edited ticket 61 to note that this could be included in the genericcard.

### CODE REVIEW

File: frontend/src/components/browse-sightings-page/BrowseSightings.scss [hunk 1]
```diff
@@ -14,19 +14,27 @@
     align-self: center;
   }
 
-  h3 {
+  .sighting-card_title {
```
Comment (TimLeach635) [inline frontend/src/components/browse-sightings-page/BrowseSightings.scss:17]: Very close! BEM dictates two underscores:
```suggestion
  .sighting-card__title {
```

File: frontend/src/components/browse-sightings-page/BrowseSightings.scss [hunk 2]
```diff
@@ -14,19 +14,27 @@
...
-    font-size: xx-large;
+    font-size: x-large;
     font-kerning: normal;
     font-weight: bold;
     padding: 1rem;
+    padding-left: 2rem;
+    margin-bottom: 0px;
+  }
+
+  .sighting-card_date {
```
Comment (TimLeach635) [inline frontend/src/components/browse-sightings-page/BrowseSightings.scss:28]:
```suggestion
  .sighting-card__date {
```

File: frontend/src/components/browse-sightings-page/SightingCard.tsx [hunk 3]
```diff
@@ -11,8 +11,19 @@ export const SightingCard: React.FunctionComponent<SightingProps> = ({
...
-      <h3 className="fade-in">Sighting #{sighting.id}</h3>
+      {!sighting.species?.name ? (
+        <h3 className="sighting-card_title fade-in">
+          Sighting recorded by {sighting.seenBy}
+        </h3>
+      ) : (
+        <h3 className="sighting-card_title fade-in">
+          Sighting of {sighting.species?.name}
+        </h3>
+      )}
```
Comment (TimLeach635) [inline frontend/src/components/browse-sightings-page/SightingCard.tsx:22]: Avoid the "`!`" and swap these two:
```suggestion
      {sighting.species?.name ? (
        <h3 className="sighting-card_title fade-in">
          Sighting of {sighting.species?.name}
        </h3>
      ) : (
        <h3 className="sighting-card_title fade-in">
          Sighting recorded by {sighting.seenBy}
        </h3>
      )}
```

File: frontend/src/components/browse-sightings-page/SightingCard.tsx [hunk 4]
```diff
@@ -11,8 +11,19 @@ export const SightingCard: React.FunctionComponent<SightingProps> = ({
...
 
+      <h6 className="sighting-card_date fade-in">
+        {new Date(sighting.seenOn).toDateString()}
+      </h6>
```
Comment (TimLeach635) [inline frontend/src/components/browse-sightings-page/SightingCard.tsx:26]: This isn't (semantically) a heading - use a `<p>` and style it
```suggestion
      <p className="sighting-card_date fade-in">
        {new Date(sighting.seenOn).toDateString()}
      </p>
```

File: frontend/src/components/browse-sightings-page/SightingCard.tsx [hunk 5]
```diff
@@ -29,11 +40,15 @@ export const SightingCard: React.FunctionComponent<SightingProps> = ({
 
       <div className="card-data fade-in">
         <p>
-          <span className="sighting-card-information">Description: </span>
+          <span className="sighting-card-information">
+            <strong>Description of sighting: </strong>
```
Comment (TimLeach635) [inline frontend/src/components/browse-sightings-page/SightingCard.tsx:44]: Don't use `<strong>` for styling purposes - it's a semantic element with a very specific use-case! If you want bold text, style the `<span>`:
```suggestion
            Description of sighting:
```

File: frontend/src/components/browse-sightings-page/SightingCard.tsx [hunk 6]
```diff
@@ -29,11 +40,15 @@ export const SightingCard: React.FunctionComponent<SightingProps> = ({
...
+          </span>
           {sighting.description}
         </p>
         <p>
-          <span className="sighting-card-information">No. whales: </span>
+          <span className="sighting-card-information">
+            <strong>Number of whales seen: </strong>
```
Comment (TimLeach635) [inline frontend/src/components/browse-sightings-page/SightingCard.tsx:50]: Ditto:
```suggestion
            Number of whales seen:
```

**Review by TimLeach635 — CHANGES_REQUESTED** [review]

File: frontend/src/components/browse-sightings-page/BrowseSightings.scss (hunk 1 above)
Comment (tomscambler) [reply frontend/src/components/browse-sightings-page/BrowseSightings.scss:17]: ok thanks, done!

File: frontend/src/components/browse-sightings-page/BrowseSightings.scss (hunk 2 above)
Comment (tomscambler) [reply frontend/src/components/browse-sightings-page/BrowseSightings.scss:28]: done, thanks!

File: frontend/src/components/browse-sightings-page/SightingCard.tsx (hunk 3 above)
Comment (tomscambler) [reply frontend/src/components/browse-sightings-page/SightingCard.tsx:22]: makes sense, thanks!

File: frontend/src/components/browse-sightings-page/SightingCard.tsx (hunk 4 above)
Comment (tomscambler) [reply frontend/src/components/browse-sightings-page/SightingCard.tsx:26]: done, thanks

File: frontend/src/components/browse-sightings-page/SightingCard.tsx (hunk 5 above)
Comment (tomscambler) [reply frontend/src/components/browse-sightings-page/SightingCard.tsx:44]: sorted, thanks!

File: frontend/src/components/browse-sightings-page/SightingCard.tsx (hunk 6 above)
Comment (tomscambler) [reply frontend/src/components/browse-sightings-page/SightingCard.tsx:50]: sorted, thanks!

File: frontend/src/components/browse-sightings-page/SightingCard.tsx [hunk 7]
```diff
@@ -11,8 +11,19 @@ export const SightingCard: React.FunctionComponent<SightingProps> = ({
...
+          Sighting of {sighting.species?.name}
+        </h3>
+      ) : (
+        <h3 className="sighting-card__title fade-in">
+          Sighting recorded by {sighting.seenBy}
+        </h3>
+      )}
 
+      <p className="sighting-card__date fade-in">
+        {new Date(sighting.seenOn).toDateString()}
```
Comment (TimLeach635) [inline frontend/src/components/browse-sightings-page/SightingCard.tsx:25]: Prefer use of [`date-fns`](https://date-fns.org/), [`format`](https://date-fns.org/v2.29.3/docs/format), and possibly [`parse`](https://date-fns.org/v2.29.3/docs/parse) (which should be already installed in `main`, so please merge from there before doing anything else!)

Also in `main` is a change that, I _think_, changes the type of `sighting.seenOn` from a `string` to a `Date()`, so you don't need to parse it from a string first!

**Review by TimLeach635 — CHANGES_REQUESTED** [review]

File: frontend/src/components/browse-sightings-page/SightingCard.tsx (hunk 7 above)
Comment (tomscambler) [reply frontend/src/components/browse-sightings-page/SightingCard.tsx]: done now, thanks

**Review by TimLeach635 — APPROVED** [review]

**Pull request merged**